src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java

```diff
@@ -110,4 +135,77 @@ protected static List<Configuration> getModuleConfigsByIds(String... moduleIds)
         return getModuleConfigsByIds(CONFIGURATION, moduleIds);
     }
 
+    /**
+     * Performs verification of the file with the given file path against the whole google config.
+     *
+     * @param filePath file path to verify.
+     * @throws Exception if exception occurs during verification process.
+     */
+    public void verifyWithWholeGoogleConfig(String filePath) throws Exception {
```

Comment:
No public methods except for tests. If something can be made `private`, then it should be that. Otherwise it should be `protected`.

I wouldn't name methods `Google` while in a `Google` abstract class. It sounds like code which calls `toString` on a `String`. Make sure to do this for others and update the javadocs.

Reply:
as we extend abstract classes, we inherit from most basic methods without "Google" in name, so in final Test class it might be confusing what method to use as it has all methods available for use. Imaginary: verifyWithWholeGoogleConfig , verifyWithItConfig, verifyWithConfig, verifyWithModules

Reply:
Following up Roman's reply... Currently we're able to access the `verifyWithItConfig` in test files. I couldn't find a way to make it not accessible there. It would be helpful if you can suggest something about this. Thanks for pointing this out

Reply:
> Currently we're able to access the verifyWithItConfig in test files. I couldn't find a way to make it not accessible there. It would be helpful if you can suggest something about this. 

There is no way to prohibit this in Java. We would have to make the method private but then `AbstractGoogle` wouldn't be able to use it. We are stuck for now.

Reply:
> I wouldn't name methods Google while in a Google abstract class. 

I still think a lot of "Google" words in this abstract should be removed. Method names and local variables.

Truthfully the only "Google" things in this class appear to be `CONFIGURATION` and `XML_NAME` and neither is named google either. It is because it is Google's configuration and Google's XML. The name is trimmed because the abstract class is Google's.

Reply:
done. removed `google` from every method and javadoc. Changed verify methods names to 

`verifyWithWholeConfig()`
`verifyWithConfigParser()` as discussed in discord

Reply:
@Zopsss This doesn't look completely done. I see one left.

> final DefaultConfiguration googleConfig = new DefaultConfiguration(ROOT_MODULE_NAME);

Reply:
@Zopsss 
> No public methods except for tests. If something can be made private, then it should be that. Otherwise it should be protected.

This is not done either. Let me know if there is some conflict.

Reply:
> @Zopsss This doesn't look completely done. I see one left.
> 
> > final DefaultConfiguration googleConfig = new DefaultConfiguration(ROOT_MODULE_NAME);

In this method, we are only loading some specific modules so we should create a config which contains only these modules and their properties. If we create the config whose root will be google_checks.xml's Checker module then we will be copying its properties to the new generated config. So should we really change this and use original Checker module of google_checks.xml as the top level module?

Reply:
I was only commenting to remove `google` from the variable name `googleConfig`. I wasn't trying to change any logic.

Reply:
oh, sorry I misunderstood. Changed the variable name now to `rootConfig`

Reply:
> @Zopsss
> 
> > No public methods except for tests. If something can be made private, then it should be that. Otherwise it should be protected.
> 
> This is not done either. Let me know if there is some conflict.
> 
> `verifyWithConfigParser` and `verifyWithWholeConfig` are both `public`.

done. made them protected